packages/noya-state/src/reducers/__tests__/layerReducer.test.ts

```diff
@@ -530,3 +530,93 @@ describe('grouping', () => {
     ]);
   });
 });
+
+describe('addLayer', () => {
+  test('selected layer is a group', () => {
+    const state = createInitialState(
+      createSketchFile(SketchModel.page({ layers: [rectangle, oval] })),
+    );
+
+    const groupedState = layerReducer(
+      state,
+      ['groupLayers', [rectangle.do_objectID, oval.do_objectID]],
+      context,
+    );
+
+    const updated = layerReducer(
+      groupedState,
+      ['addLayer', rectangle],
+      context,
+    );
+
+    expect(
+      debugDescription([
+        Selectors.getCurrentPage(state),
+        Selectors.getCurrentPage(updated),
+      ]),
+    ).toMatchSnapshot();
+  });
+  test('original layer is in parent layer', () => {
```

Comment:
What do we do differently if the original layer is in the parent? Does the parent need to be selected? I'm not sure I understand this one

Reply:
Depends on which layer it's selected. 
1. If a selected layer it's a parent layer but not a page (group, artboard, etc) the pasted element would be added to that layer at the end. 
2. If the parent layer it's a page and the original layer it's on the page we'll add it be under that element
3. If the original layer is not on the page and we have a layer selected, it'll add it under that layer.
4. If we don't have any layer selected just the page it will just add it at the end

I hope that makes sense 😅

Reply:
Got it! I'll have to play with it some more to see if that feels right, but very helpful to have the tests to demonstrate that those different cases are working correctly